src/Service/SF1500/SF1500.php

```diff
@@ -0,0 +1,859 @@
+
+                if (null === $adresseId) {
+                    continue;
+                }
+
+                $data = $this->adresseLaes($adresseId, $token);
+
+                return $this->getValue($data, $adresseTekstKeys, '');
+            }
+        }
+
+        return '';
+    }
+
+    /**
+     * Fetches person magistrat from SF1500.
+     */
+    public function getPersonMagistrat(string $funktionsId)
+    {
+        $orgEnhedId = $this->getOrganisationEnhed($funktionsId, true);
+
+        if (empty($orgEnhedId)) {
+            return '';
+        }
+
+        $token = $this->fetchSAMLToken();
+
+        if (null === $token) {
+            return '';
+        }
+
+        $data = $this->organisationEnhedLaes($orgEnhedId, $token);
+
+
+        $enhedsNavnKeys = [
+            'ns3LaesOutput',
+            'ns3FiltreretOejebliksbillede',
+            'ns3Registrering',
+            'ns3AttributListe',
+            'ns3Egenskab',
+            'ns2EnhedNavn',
+        ];
+
+        $enhedNavn = $this->getValue($data, $enhedsNavnKeys);
+
+        $data = $this->getEnhedNavnOgOverordnetOrganisationsId($orgEnhedId, $token);
+
+        while ($orgEnhedId = $data['overordnet_id']) {
+            $enhedNavn = $data['enhedNavn'];
+            $data = $this->getEnhedNavnOgOverordnetOrganisationsId($orgEnhedId, $token);
+        }
+
+        return $enhedNavn;
+    }
+
+    /**
+     * Fetches enhedsnavn and overordnet organisations id.
+     */
+    public function getEnhedNavnOgOverordnetOrganisationsId($organisationEnhedsId, $token): array
+    {
+        $data = $this->organisationEnhedLaes($organisationEnhedsId, $token);
+
+        $overordnetKeys = [
+            'ns3LaesOutput',
+            'ns3FiltreretOejebliksbillede',
+            'ns3Registrering',
+            'ns3RelationListe',
+            'ns2Overordnet',
+            'ns2ReferenceID',
+            'ns2UUIDIdentifikator',
+        ];
+
+        $enhedsNavnKeys = [
+            'ns3LaesOutput',
+            'ns3FiltreretOejebliksbillede',
+            'ns3Registrering',
+            'ns3AttributListe',
+            'ns3Egenskab',
+            'ns2EnhedNavn',
+        ];
+
+        return [
+            'overordnet_id' => $this->getValue($data, $overordnetKeys),
+            'enhedNavn' => $this->getValue($data, $enhedsNavnKeys),
+        ];
+    }
+
+    /**
+     * Fetches bruger and organisation funktions id for nearest manager or null if no manager exists.
+     */
+    public function getNearestManagerBrugerAndFunktionsId($userId, $managerFunktionsTypeId)
+    {
+        $token = $this->fetchSAMLToken();
+
+        if (null === $token) {
+            return '';
+        }
+
+        $organisationFunktionsId = $this->getOrganisationFunktionerFromUserId($userId);
+
+        // Select just one of the organisation funktioner(ansættelser).
+        if (is_array($organisationFunktionsId)) {
+            $organisationFunktionsId = reset($organisationFunktionsId);
+        }
```

Comment:
Why? We must be able to handle multiple “ansættelser” if a “bruger” can have multiple.

Reply:
Refactored to handle multiple.